Heat dissipation (#57)
LibreSolar/learn.libre.solar: docs/development/heat_dissipation.md

# Losses and Heat

> In every real electrical system, flowing currents are generating heat inside the components. While most low energy devices do not generate siginificant amount of thermal waste, power electronics need to dissipate large amount of thermal energy.

Reviewer: ```suggestion
In every real electrical system, flowing currents are generating heat inside the components. While most low energy devices do not generate significant amount of losses, power electronics need to dissipate large amount of thermal energy.
```

There are many different components on a board that contribute to the overall power loss, like resistors, traces, fuses, coil resistance and other ohmic resistances. In circuits with frequent switching, the impedance of capacitators and inductances can play a role as well, but won't be treated here.

> The main factor for heat generation in power electronics are the semi-conductors, mainly **MOSFETS** used for switching. In the following, switching and conduction losses for **MOSFETs** are explained and different ways to dissipate the heat are given.

Reviewer: ```suggestion
The main factor for heat generation in power electronics are the semi-conductors, mainly **MOSFETs** used for switching. In the following, switching and conduction losses for MOSFETs are explained and different ways to dissipate the heat are given.
```

## Conduction loss

Conduction loss appears in the steady-state phase when the system is not changing anymore after being switched on. The power can be calculated using

> $$ P_{cond} = r_{ds} \cdot I^2 $$

Reviewer: - Correct symbol for heat is Q, not P (even though the unit is the same).
- R in capital letters. On-state resistance typically written as below.
(please fix also in other locations)
```suggestion
$$ Q_{cond} = R_{DS(on)} \cdot I^2 $$
```

where $r_{ds}$ is the internal gate to source resistance of the transistor, which can be obtained from the datasheet and $I$ is the current flowing through it. If the transitor is used in converters or else and is switched on and off frequently (e.g. a PWM [charge controller](../system/charge_controller.md)), the duty cycle (in percent) should be used to weigh the power loss:

$$ P_{cond} = r_{ds} \cdot I^2 \cdot D$$

> To reduce conduction loss, a bigger transistor with a larger channel can be used to reduce $r_{ds}$. That also means a longer switching time because the capacitance in the transistor is greater, which increases the switching time and therefor the switching loss. In high frequency switching applications with a low duty cycle, it can be better to optimize for low switching time than low internal resistance.

Reviewer: ```suggestion
To reduce conduction loss, a bigger transistor with a larger channel can be used to reduce $r_{ds}$. That also means a longer switching time because the internal gate capacitance in the transistor is greater, which increases the switching time and therefore the switching loss. In high frequency switching applications with a low duty cycle, it can be better to optimize for low switching time than low internal resistance.
```

## Switching loss

When a transistor is switching, the conducting channel has to build up. When the gate to source voltage is higher than the threshold voltage, the channels starts to form and the resistance $R_{DS}$ starts to fall. Until the channel has formed completly, the transistor acts as a ohmic resistance, the power loss in it can be obtained by

> $$P_{swit} = \frac{1}{2\cdot T_s} \cdot t_{on} \cdot U_{ds} \cdot I_{ds}$$

Reviewer: All heat flows should be `Q` as mentioned before (not `P`).

Also please use capital DS for drain-source voltage and current so that it's the same as for the resistor.

Final remark I realized just now: In English, `V` is the typical symbol for voltages and not `U`. In order to be consistent with datasheets of MOSFETs (see e.g. [here](https://www.mouser.de/datasheet/2/196/Infineon_IRFP4568_DataSheet_v01_01_EN-1228352.pdf)) I'd ask you to update this aswell.

Reviewer: Ooops, sorry. The heat flow is actually the derivative of Q, so it must be Q with a dot. Seems like [this page](https://en.wikipedia.org/wiki/Thermal_resistance) on Wikipeda is wrong, as it uses Q without a dot all the time...

Reviewer: Wikipedia page is now fixed. `\dot{Q}` is correct for heat flow.

where $T_s$ is the timespan for one switching process. The equivalent function with $f_s = \frac{1}{T_s}$ shows that the switching loss increases with the switching frequency:

$$P_{swit} = \frac{1}{2} \cdot t_{on} \cdot U_{ds} \cdot I_{ds} \cdot f_s$$

Both $U_{ds}$ and $I_{ds}$ are changing during the switching time. Figure 1 shows voltage and current during the process.

<figure>
    <switching-loss/>
    <br/>
<center>
    <figcaption><b>Figure 1.</b> Current and voltage of the transistor during switching.
    </figcaption>
</center>
</figure>

At $t1$, the gate voltage reaches the threshold voltage and the drain-source current starts to flow. At $t2$ the current reaches its maximum and the drain voltage starts to fall. In this point, the switching power loss is at its maximum too. Until $t3$, the drain voltage drops as inner resistance $r_{ds}$ decreases and so $t_{on} = t_3 -t_1$. The same procedure in reverse happens when the MOSFET is switched off.

The total power loss per duty cylce can be aggregated as

$$P_{tot} = 2\cdot P_{swit} + P_{cond}$$

## Heat dissipation

> To dissipate heat from the board, some design considerations have to be made. In general, heat can be transfered through convection, conduction and radiation where radiation is the least effective and only used in special cases. Dissipation is necessary to keep parts from overheating and especially semiconductors deteriorate faster when working under high temperatures.

Reviewer: ```suggestion
To dissipate heat from the board, some design considerations have to be made. In general, heat can be transferred through convection, conduction and radiation, where radiation is the least effective at low temperatures. Dissipation is necessary to keep parts from overheating. Especially semiconductors deteriorate faster when working under high temperatures.
```

> In many cases, conduction is used to transfer heat between parts, e.g. the MOSFET case and a heat sink while convection is used to transfer the heat to the environment and away from the component. The main factor for heat dissipation is the thermal resistance. This can be the connection from a MOSFET to a heat sink or the junction within the MOSFET to its case. It depends on the thermal conductivity of elements and connection between parts. The greater the surface of a connection, the lower the resistance, copper has a better conductivity thus lower thermal resistance than plastic etc. For many components, thermal resistance is given in the datasheet.

Reviewer: ```suggestion
In many cases, conduction is used to transfer heat between parts, e.g. the MOSFET case and a heat sink while convection is used to transfer the heat to the environment and away from the component. The main factor for heat dissipation is the thermal resistance. This can be the connection from a MOSFET to a heat sink or the junction within the MOSFET to its case. It depends on the thermal conductivity of elements and connection between parts. The greater the surface of a connection, the lower the resistance. For example, copper has a better conductivity and thus lower thermal resistance than plastic. For many electronic components, thermal resistance values are given in the datasheet.
```

The calculations for thermal resistance and heat transfer are quite similar to those for voltage, current and resistance.

The thermal resistance is given in °C/W, the temperature in °C and the power in W.
When a power (equivalent to current) of 1W is flowing through a thermal resistance with 2°C/W (equivalent to electrical resistance), a temperature difference of 2°C is seen over the component (equivalent to voltage drop across a resistance). Like with electronic circuits, thermal resistances can be placed in series or in parallel.

**Example:**

> A heat sink is connected to a MOSFET with a screw only (no heat paste or silikon conductor). The thermal resistance between air and heat sink is $R_{th,1}=6$°C/W and the connection with the MOSFET has a thermal resistance of $R_{th,2}=1$°C/W. The connection between the semiconductor ant the casing has a thermal resistance of $R_{th,3}=5$°C/W and the ambient temperature is 20°C. The power dissipated in the semiconductor is $P=5$W.

Reviewer: ```suggestion
A heat sink is connected to a MOSFET with a screw only (no heat paste or silikon conductor). The thermal resistance between air and heat sink is $R_{th,1}=6$°C/W and the connection with the MOSFET has a thermal resistance of $R_{th,2}=1$°C/W. The connection between the semiconductor and the casing has a thermal resistance of $R_{th,3}=5$°C/W and the ambient temperature is 20°C. The power dissipated in the semiconductor is $P=5$W.
```

The first temperature difference from the air to the heat sink can be calculated:

$$\Delta T_1 = P \cdot R_{th, 1} = 5W \cdot 6^\circ C/W = 30^\circ C$$

The next temperature differences are calculated accordingly:

$$\Delta T_2 = P \cdot R_{th, 2} = 5W \cdot 1^\circ C/W = 5^\circ C \\
\Delta T_3 = P \cdot R_{th, 3} = 5W \cdot 5^\circ C/W = 25^\circ C$$

So beginning with the ambient temperature, the epected temperature inside the MOSFET is the sum of the differences across all thermal resistances:

$$T_{trans} = T_{amb} + \Delta T_1 + \Delta T_2 + \Delta T_3 = 80^\circ C$$

It is clear, that the temperature rises with the ambient temperature or with increased thermal resistance in the chain. While this method is not highly precise, it gives a good overview of the weakest links and can help to improve cooling. The goal should always be to keep the temperature as low as possible without exaggerated complexity and cost.

### Passive cooling

> When passive cooling is used, the air flow (or any other liquid) is created simply by the density differences due to temperature gradients. The MOSFET can be either connected to the PCB where its surface is than passing the heat to the air passively or to heat sink. The heat sink has an optimized surface and good heat conducting characteristics to spread the heat fast and transfer it to the air over a large surface area. In both cases, the airflow is low and so is the amount of heat dissipated. For low energy appliances, this method is suitable, for larger power electronics, active cooling can be necessary.

Reviewer: ```suggestion
When passive cooling is used, the air flow (or any other liquid) is created simply by the density differences due to temperature gradients. The MOSFET can be either connected to the PCB where its surface is then passing the heat to the air passively or to a heat sink. The heat sink has an optimized surface and good heat conducting characteristics to spread the heat fast and transfer it to the air over a large surface area. In both cases, the airflow is low and so is the amount of heat dissipated. For low power appliances, this method is suitable, for larger power electronics, active cooling can be necessary.
```

### Active Cooling

> With active cooling, a forced flow is generated over the surface of the PCB or heat sink. This can be done by simply blowing air with a fan or pumping a liquid. While a fan is mostly the cheapes and easiest to implement, using a water cooled system can be necessary for very high power density and with limited space.

Reviewer: ```suggestion
With active cooling, a forced flow is generated over the surface of the PCB or heat sink. This can be done by simply blowing air with a fan or pumping a liquid. While a fan is mostly cheapest and easiest to implement, using a water cooled system can be necessary for very high power density and with limited space.
```

In the above example, if active cooling would reduce the thermal resistance of the heat sink by e.g. 50%, the expected temperature would be reduced by 15°C.

> ### Further onformation

Reviewer: ```suggestion
### Further information
```
